sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -0,0 +1,290 @@
+        # estimator type
+        # --------------
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "scitype:y": "univariate",
+        "requires-fh-in-fit": True,
+        "X-y-must-have-same-index": True,
+        "handles-missing-data": False,
+        "capability:insample": False,
+    }
+
+    def __init__(
+        self: "_NeuralForecastAdapter",
+        freq: str,
+        local_scaler_type: typing.Optional[
+            typing.Literal["standard", "robust", "robust-iqr", "minmax"]
+        ] = None,
+        futr_exog_list: typing.Optional[typing.List[str]] = None,
+        verbose_fit: bool = False,
+        verbose_predict: bool = False,
+    ) -> None:
+        self.freq = freq
+        self.local_scaler_type = local_scaler_type
+
+        self.futr_exog_list = futr_exog_list
+
+        self.verbose_fit = verbose_fit
+        self.verbose_predict = verbose_predict
+
+        super().__init__()
+
+        self.id_col = "unique_id"
+        self.time_col = "ds"
+        self.target_col = "y"
+
+        self.needs_X = self.algorithm_exogenous_support and bool(self.futr_exog_list)
+
+        self.set_tags(**{"ignores-exogeneous-X": not self.needs_X})
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_exogenous_support(self: "_NeuralForecastAdapter") -> bool:
+        """Set support for exogenous features."""
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_name(self: "_NeuralForecastAdapter") -> str:
+        """Set custom model name."""
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_class(self: "_NeuralForecastAdapter"):
+        """Import underlying NeuralForecast algorithm class."""
+
+    @functools.cached_property
+    @abc.abstractmethod
+    def algorithm_parameters(self: "_NeuralForecastAdapter") -> dict:
+        """Get keyword parameters for the underlying NeuralForecast algorithm class.
+
+        Returns
+        -------
+        dict
+            keyword arguments for the underlying algorithm class
+
+        Notes
+        -----
+        This method should not include following parameters:
+
+        - future exogenous columns (``futr_exog_list``) - used from ``__init__``
+        - historical exogenous columns (``hist_exog_list``) - not supported
+        - statis exogenous columns (``stat_exog_list``) - not supported
+        - custom model name (``alias``) - used from ``algorithm_name``
+        """
+
+    def _instantiate_model(self: "_NeuralForecastAdapter", fh: ForecastingHorizon):
+        """Instantiate the model."""
+        exogenous_parameters = (
+            {"futr_exog_list": self.futr_exog_list} if self.needs_X else {}
+        )
+
+        algorithm_instance = self.algorithm_class(
+            fh,
+            alias=self.algorithm_name,
+            **self.algorithm_parameters,
+            **exogenous_parameters,
+        )
+
+        from neuralforecast import NeuralForecast
```

Comment:
This takes frequency as init argument.

> a bit unsettling that freq is an argument of __init__, rather than obtained from X or fh. Is the latter not possible?

I won't say impossible, but with the amount of issues in `pandas` frequency specifications, and what @benHeid faced in the other issue, personally I'll feel safer (definitely easier) for leaving this choice to user to configure rather than adding a frequency inference logic in `sktime`.

Reply:
yes, but what I mean, can you not just take it from `fh`? I.e., instea dof the `__init__` in `_fit`, you take `fh.freq`.

What happens, say, if the `freq` in `y` and that of the `__init__` do not agree? Does that not lead to wrong behaviour?

Reply:
The `freq` is just used to build the future timestamps when calling `predict`. Since you don't use those timestamps you could just set `freq='D'` internally, for example.

Reply:
> The freq is just used to build the future timestamps when calling predict. Since you don't use those timestamps you could just set freq='D' internally, for example.

Hi @jmoralez, I am unable to follow how passing a fixed freuqency, e.g. `D` will suffice. I posted an example above (direct use of neuralforecast), where it seems to generate wrong dates if frequency is incorrect. Can you please elaborate?

Reply:
I saw that you only extract the prediction column (`model_point_predictions = model_forecasts[prediction_column_names[0]].to_numpy()`) from the dataframe that neuralforecast returns, so I don't think the timestamps it returns matter, do they?

Those timestamps are there just to make it easier for a user to merge with a validation set or similar, the predictions will be the same if you set different frequencies (only the timestamps will change).

Reply:
Okay, let me fix it at `D` and trigger all tests.

Reply:
Oh, sorry. We do check the timestamps of the `futr_df` when calling predict, that's my bad. Those timestamps need to match the ones generated with the specified `freq` from `__init__`